File: БР1.1/Захматов Юрий/labs/lab2/booking-service/src/main/java/com/petproject/bookingservice/controllers/BookingController.java
package com.petproject.bookingservice.controllers;

import com.petproject.bookingservice.dto.BookingCreateRequest;
import com.petproject.bookingservice.dto.BookingResponse;
import com.petproject.bookingservice.security.JwtPrincipal;
import com.petproject.bookingservice.service.BookingService;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.springframework.data.domain.Page;
import org.springframework.http.ResponseEntity;
import org.springframework.security.core.annotation.AuthenticationPrincipal;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api/v1/properties/{propertyId}/bookings")
@RequiredArgsConstructor
public class BookingController {

    private final BookingService bookingService;

    @PostMapping
    public ResponseEntity<BookingResponse> createBooking(
            @PathVariable("propertyId") Long propertyId,
            @AuthenticationPrincipal JwtPrincipal user,
            @RequestBody @Valid BookingCreateRequest request
            ) {
        var response = bookingService.createBooking(propertyId, request, user);
        return ResponseEntity.ok(response);
    }


    @GetMapping
    public ResponseEntity<Page<BookingResponse>> getAllBookings(
            @PathVariable("propertyId") Long propertyId,
            @RequestParam(defaultValue = "0") int page,
            @RequestParam(defaultValue = "20") int size
    ) {
        var response = bookingService.getAllBookings(propertyId, page, size);
        return ResponseEntity.ok(response);
    }

    @GetMapping("/{id}")
    public ResponseEntity<BookingResponse> getBooking(
            @PathVariable("propertyId") Long propertyId,
            @PathVariable("id") Long id,
            @AuthenticationPrincipal JwtPrincipal user
    ) {
        var response = bookingService.getBookingById(propertyId, id, user);
        return ResponseEntity.ok(response);
    }

    // только арендующий
    @PatchMapping("/{id}/cancel")
    public ResponseEntity<BookingResponse> cancelBooking(
            @PathVariable Long id,
            @AuthenticationPrincipal JwtPrincipal user
    ) {
        BookingResponse response = bookingService.cancelBooking(id, user);
        return ResponseEntity.ok(response);
    }

    // только арендодатель
    @PatchMapping("/{id}/confirm")
    public ResponseEntity<BookingResponse> confirmBooking(
            @PathVariable Long id,
            @AuthenticationPrincipal JwtPrincipal user
    ) {
        BookingResponse response = bookingService.confirmBooking(id, user);
        return ResponseEntity.ok(response);
    }
    // только арендодатель
    @PatchMapping("/{id}/reject")
    public ResponseEntity<BookingResponse> rejectBooking(
            @PathVariable Long id,
            @AuthenticationPrincipal JwtPrincipal user
    ) {
        BookingResponse response = bookingService.rejectBooking(id, user);
        return ResponseEntity.ok(response);
    }

    // только арендодатель (только позже даты окончания)
    @PatchMapping("/{id}/complete")
    public ResponseEntity<BookingResponse> completeBooking(
            @PathVariable Long id,
            @AuthenticationPrincipal JwtPrincipal user
    ) {
        BookingResponse response = bookingService.completeBooking(id, user);
        return ResponseEntity.ok(response);
    }
}

File: БР1.2/Цатинян Артём/labs/lab2/booking-service/src/main/kotlin/ru/itmo/restaurantbooking/lab2/booking/adapter/rest/BookingController.kt
package ru.itmo.restaurantbooking.lab2.booking.adapter.rest

import io.swagger.v3.oas.annotations.Parameter
import jakarta.validation.Valid
import jakarta.validation.constraints.Max
import jakarta.validation.constraints.Min
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PatchMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.ResponseStatus
import org.springframework.web.bind.annotation.RestController
import ru.itmo.restaurantbooking.lab2.booking.adapter.rest.dto.CreateBookingRequest
import ru.itmo.restaurantbooking.lab2.booking.service.BookingService
import ru.itmo.restaurantbooking.lab2.common.auth.AuthenticatedUser
import java.time.LocalDate

@RestController
@RequestMapping("/api/v1")
class BookingController(
    private val bookingService: BookingService
) {
    @GetMapping("/restaurants/{restaurantId}/availability")
    fun availability(
        @PathVariable restaurantId: Long,
        @RequestParam date: LocalDate,
        @RequestParam(defaultValue = "2") @Min(1) @Max(20) guestsCount: Int
    ) = bookingService.availability(restaurantId, date, guestsCount)

    @PostMapping("/bookings")
    @ResponseStatus(HttpStatus.CREATED)
    fun create(
        @Parameter(hidden = true)
        currentUser: AuthenticatedUser,
        @Valid @RequestBody request: CreateBookingRequest
    ) = bookingService.create(currentUser.id, request)

    @GetMapping("/bookings/me")
    fun mine(
        @Parameter(hidden = true)
        currentUser: AuthenticatedUser,
        @RequestParam(defaultValue = "1") page: Int,
        @RequestParam(defaultValue = "10") size: Int
    ) = bookingService.mine(currentUser.id, page.coerceAtLeast(1), size.coerceIn(1, 100))

    @GetMapping("/bookings/{bookingId}")
    fun byId(
        @Parameter(hidden = true)
        currentUser: AuthenticatedUser,
        @PathVariable bookingId: Long
    ) = bookingService.byId(currentUser.id, bookingId)

    @PatchMapping("/bookings/{bookingId}/cancel")
    fun cancel(
        @Parameter(hidden = true)
        currentUser: AuthenticatedUser,
        @PathVariable bookingId: Long
    ) = bookingService.cancel(currentUser.id, bookingId)
}
